lib/ts/recipe/oauth2/types.ts

```diff
@@ -0,0 +1,179 @@
+
+        // OpenIdConnect
+        prompt?: string;
+        nonce?: string;
+
+        // Not sure if we want to support these
+        acr_values?: string;
+        maxAge?: number;
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<{
+        redirectUrl: string; // This will end up as a 302, encoding both error and success
+    }>;
+    authorizePOST(input: {
+        clientId: string;
+        responseType: string;
+        redirectUri?: string;
+        scopes: string[];
+        state?: string;
+
+        // PKCE
+        codeChallenge?: string;
+        codeChallengeMethod?: string;
+
+        // OpenIdConnect
+        prompt?: string;
+        nonce?: string;
+
+        // Not sure if we want to support these
+        acr_values?: string;
+        maxAge?: number;
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<{
+        redirectUrl: string; // This will end up as a 302, encoding both error and success
+    }>;
+    // This is intended to be used when we want to do the redirection in the FE (e.g.: header based auth)
+    getRedirectUrlPOST(input: {
+        clientId: string;
+        responseType: string;
+        redirectUri?: string;
+        scopes: string[];
+        state?: string;
+
+        // PKCE
+        codeChallenge?: string;
+        codeChallengeMethod?: string;
+
+        // OpenIdConnect
+        prompt?: string;
+        nonce?: string;
+
+        // Not sure if we want to support these
+        acr_values?: string;
+        maxAge?: number;
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<
+        | {
+              status: "OK";
+              redirectUrl: string;
+          }
+        | GeneralErrorResponse
+    >;
+
+    userInfoGET(input: {
+        options: APIOptions;
+        userContext: any;
+    }): Promise<{ status: "OK"; sub: string; email?: string; email_verified?: boolean; phoneNumber?: string }>; // We should define this in a way that doesn't require adding all claims here
+};
+
+export type APIInterface = {
+    buildAccessToken(input: {
+        clientId: string;
+        userId?: string;
+        sessionHandle?: string;
+        scopes: string[];
+    }): Promise<{ payload?: JSONObject; lifetimeInSecs?: number }>; // Returning undefined for these props means we use the default from Core
+    buildIdToken(input: {
+        clientId: string;
+        userId?: string;
+        sessionHandle?: string;
+        scopes: string[];
+    }): Promise<{ payload?: JSONObject; lifetimeInSecs?: number }>; // Returning undefined for these props means we use the defaults from BE SDKs
+    getRefreshTokenLifetime(): Promise<number | undefined>; // Returning undefined means we use the default from Core
+
+    buildAuthorizeRedirectUrl(input: {
+        clientId: string;
+        redirectUri: string;
+        error?: string;
+        code?: string;
+        idToken?: string;
+    }): Promise<string>;
+
+    validateScopes(input: {
```

Comment:
We've discussed this on the call, but:
- We return true for some built-in scopes (`supertokens.profile`, `openid`)
- Anything else is denied (not removed from the list, since it's easier to debug if it fails loudly)